### ISSUE TITLE
Add Polar Express Muon

### ISSUE THREAD

**Issue by shermansiu** [microsoft/dion#26]
According to the following paper, using Polar Express instead of Newton-Schultz in Muon leads to improvements in the validation loss.
https://arxiv.org/abs/2505.16932

It was also used in Karpathy's NanoChat repo to train a GPT-2 equivalent model for less than $100 (https://github.com/karpathy/nanochat/discussions/481)

It might be worthwhile to add it here.

**Official implementation:** https://github.com/NoahAmsel/PolarExpress/blob/main/polar_express.py

**Comment by shermansiu** [comment]
And for reference, here's the implementation in NanoChat: https://github.com/karpathy/nanochat/blob/16b8ac7da33010dc7efcd9292f895703f5cff33a/nanochat/optim.py#L110-L123

**Comment by tyler-romero** [comment]
You should be able to pass this to the `newton_schulz_func` arg of `Muon`:

```
import torch

@torch.compile(fullgraph=True)
def polar_express5(G: torch.Tensor, epsilon: float = 1e-7) -> torch.Tensor:
    # https://github.com/karpathy/nanochat/blob/16b8ac7da33010dc7efcd9292f895703f5cff33a/nanochat/optim.py#L110-L123
    polar_express_coeffs = [
        (8.156554524902461, -22.48329292557795, 15.878769915207462),
        (4.042929935166739, -2.808917465908714, 0.5000178451051316),
        (3.8916678022926607, -2.772484153217685, 0.5060648178503393),
        (3.285753657755655, -2.3681294933425376, 0.46449024233003106),
        (2.3465413258596377, -1.7097828382687081, 0.42323551169305323),
    ]

    assert G.ndim >= 2
    X = G.bfloat16()
    if G.size(-2) > G.size(-1):
        X = X.mT

    X = X / (X.norm(dim=(-2, -1), keepdim=True) * 1.01 + epsilon)

    for a, b, c in polar_express_coeffs:
        A = X @ X.mT
        B = b * A + c * A @ A
        X = a * X + B @ X  # X <- aX + bX^3 + cX^5

    if G.size(-2) > G.size(-1):
        X = X.mT
    return X
```

**Comment by shermansiu** [comment]
I actually used a safety factor of 1.02 in my nanochat Polar Express implementation! It still didn't outperform the paper one

**Comment by JohnLangford** [comment]
This is merged in now, thanks :-)   My tests so far don't reveal a substantial advantage over the older coefficients so I'm planning to so some more extensive testing.
